src/persistence/existingNamespaceUpdateInterceptor.js

```diff
@@ -6,11 +6,36 @@ export default function existingNamespaceUpdateInterceptor(openmct, usersNamespa
             return shouldCheck;
         },
         invoke: (identifier, object) => {
+            // check for old user namespace style
             if (object.location === usersNamespace.key) {
                 object.location = usersNamespace.id;
                 openmct.objects.mutate(object, 'location', usersNamespace.id);
             }
 
+            // check for old composition style
+            if (object.composition?.length) {
+                let updatedComposition = false;
+
+                object.composition = object.composition.map((keystring) => {
+                    if (typeof keystring === 'string') {
+                        updatedComposition = true;
+
+                        const parts = keystring.split(':', 2);
```

Comment:
i think this could unintentionally throw away part of the string in the case of dictionary items, aka `namespace='vista'` items. Would `openmct.parseKeyString` work here?

Reply:
nail on the head my friend... will update